### ISSUE TITLE
[Build]  Cuda 12.5

### ISSUE THREAD

**Issue by johnnynunez** [microsoft/onnxruntime#20765]
### Describe the issue

```bash
[ 80%] Building CUDA object CMakeFiles/onnxruntime_providers_cuda.dir/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/quantization/dequantize_blockwise_bnb4.cu.o
/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu(68): error: identifier "FLT_MAX" is undefined
      float threadData(-FLT_MAX);
                        ^

1 error detected in the compilation of "/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu".
gmake[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:6984: CMakeFiles/onnxruntime_providers_cuda.dir/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu.o] Error 2
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:2131: CMakeFiles/onnxruntime_providers_cuda.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
Traceback (most recent call last):
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 2950, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 2842, in main
    build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 1731, in build_targets
    run_subprocess(cmd_args, env=env)
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 861, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/johnny/Projects/onnxruntime/tools/python/util/run.py", line 49, in run
    completed_process = subprocess.run(
                        ^^^^^^^^^^^^^^^
  File "/home/johnny/miniconda3/envs/py311/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/johnny/miniconda3/envs/py311/bin/cmake', '--build', '/home/johnny/Projects/onnxruntime/build/Linux/Release', '--config', 'Release', '--', '-j12']' returned non-zero exit status 2.
```

### Target platform

arm64

### Build script
```bash
git clone --recursive https://github.com/microsoft/onnxruntime
cd onnxruntime 

export PATH="/usr/local/cuda/bin:${PATH}"
export CUDACXX="/usr/local/cuda/bin/nvcc"

pip3 install -U packaging
./build.sh --config Release --update --parallel --build --build_wheel --build_shared_lib --skip_tests \
--use_tensorrt --cuda_home /usr/local/cuda --cudnn_home /usr/lib/aarch64-linux-gnu \
--tensorrt_home /usr/lib/aarch64-linux-gnu --cmake_extra_defines CMAKE_CXX_FLAGS="-Wno-unused-variable -I/usr/local/cuda/include" --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES="87"
```
### Error / output

```bash
[ 80%] Building CUDA object CMakeFiles/onnxruntime_providers_cuda.dir/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/quantization/dequantize_blockwise_bnb4.cu.o
/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu(68): error: identifier "FLT_MAX" is undefined
      float threadData(-FLT_MAX);
                        ^

1 error detected in the compilation of "/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu".
gmake[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/build.make:6984: CMakeFiles/onnxruntime_providers_cuda.dir/home/johnny/Projects/onnxruntime/onnxruntime/contrib_ops/cuda/moe/ft_moe/moe_kernel.cu.o] Error 2
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:2131: CMakeFiles/onnxruntime_providers_cuda.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
Traceback (most recent call last):
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 2950, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 2842, in main
    build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 1731, in build_targets
    run_subprocess(cmd_args, env=env)
  File "/home/johnny/Projects/onnxruntime/tools/ci_build/build.py", line 861, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/johnny/Projects/onnxruntime/tools/python/util/run.py", line 49, in run
    completed_process = subprocess.run(
                        ^^^^^^^^^^^^^^^
  File "/home/johnny/miniconda3/envs/py311/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/johnny/miniconda3/envs/py311/bin/cmake', '--build', '/home/johnny/Projects/onnxruntime/build/Linux/Release', '--config', 'Release', '--', '-j12']' returned non-zero exit status 2.
```

### GCC / Compiler Version

3.29.3

**Comment by johnnynunez** [comment]
maybe needs #include <cfloat> ??

**Comment by tianleiwu** [comment]
> maybe needs `#include <cfloat>` ??

I think so.

**Comment by johnnynunez** [comment]
> > maybe needs `#include <cfloat>` ??
> 
> I think so.

it is working

**Comment by tianleiwu** [comment]
The issue is fixed in 1.18.1 and main branches.
